File: src/test/java/org/dicio/skill/SkillTest.java
package org.dicio.skill;

import static org.junit.Assert.assertTrue;

import org.dicio.skill.chain.InputRecognizer;
import org.junit.Test;

import java.util.List;

public class SkillTest {

    public static Skill buildEmptySkill() {
        return new Skill() {
            @Override public InputRecognizer.Specificity specificity() { return null; }
            @Override public void setInput(final String input, final List<String> inputWords,
                                           final List<String> normalizedWordKeys) {}
            @Override public float score() { return 0; }
            @Override public void processInput() {}
            @Override public void generateOutput() {}
            @Override public void cleanup() {}
        };
    }

    @Test
    public void testConstructor() {
        final Skill skill = buildEmptySkill();

        assertTrue(skill.nextSkills().isEmpty());
    }
}

File: src/test/java/org/dicio/skill/SkillTest.kt
package org.dicio.skill

import org.dicio.skill.chain.InputRecognizer.Specificity
import org.junit.Assert
import org.junit.Test

class SkillTest {
    @Test
    fun testConstructor() {
        val skill = buildEmptySkill()

        Assert.assertTrue(skill.nextSkills().isEmpty())
    }

    companion object {
        fun buildEmptySkill(): Skill {
            return object : Skill() {
                override fun specificity(): Specificity {
                    return Specificity.MEDIUM
                }

                override fun setInput(
                    input: String, inputWords: List<String>,
                    normalizedWordKeys: List<String>
                ) {
                }

                override fun score(): Float {
                    return 0.0f
                }

                override fun processInput() {}
                override fun generateOutput() {}
                override fun cleanup() {}
            }
        }
    }
}
